portfolio/src/main/webapp/scripts/featured_works.js

```diff
@@ -0,0 +1,145 @@
+  const projectsHtml = projectDivs.map(projectDiv => projectDiv.outerHTML)
+                          .join('');
+  return projectsHtml;
+}
+
+/** 
+ * Converts an array of projects that is represented as JSON objects to Projects. 
+ * @return {Array<Project>}
+ */
+const toProjects = (projects_json) => {
+  const projects = projects_json.map(project_json => toProject(project_json));
+  return projects;
+}
+
+/** 
+ * Converts a JSON-represented project to Project. 
+ * @return {Project}
+ */
+const toProject = (project_json) => {
+  const name = project_json.name;
+  const image = project_json.image;
+  const description = project_json.description;
+  const tools = project_json.tools;
+  const date = project_json.date;
+  const page = project_json.page;
+  return new Project(name, image, description, tools, date, page);
+}
+
+/**
+ * Adds navigation buttons.
+ * @return {string} This string is the complete HTML content required for the
+ *    featured works section.
+ */
+function addNavButtons(projectsHtml) {
+  const leftButton = '<div id="left-button" class="button">' + 
+      '<i class="fas fa-chevron-circle-left fa-lg"></i></div>';
+  const rightButton = '<div id="right-button" class="button">' + 
+      '<i class="fas fa-chevron-circle-right fa-lg"></i></div>';
+  
+  return leftButton + projectsHtml + rightButton;
+}
+
+const displayFirstProj = () => {
+  const projects = document.getElementsByClassName('project');
+  ensureNonNull(projects);
+  projects[0].style.display = 'flex';
+}
+
+/** 
+ * Sets up the navigation of projects in the featured works section using
+ * the left and right buttons.
+ */
+function setupProjectNav() {
+  const leftButton = document.getElementById('left-button');
+  ensureNonNull(leftButton);
+  leftButton.onclick = () => moveToProj(-1);
+
+  const rightButton = document.getElementById('right-button');
+  ensureNonNull(rightButton);
+  rightButton.onclick = () => moveToProj(1);
+
+  const projects = document.getElementsByClassName('project');
+  ensureNonNull(projects);
+  /** @type {number} Indicates the index of the project that is currently shown. */
+  let projIndex= 0;
+
+  /** 
+   * Moves to the next project. If `moveBy` is negative, moves to the previous project;
+   * otherwise, if `moveBy` is positive, moves to the next project.
+   * 
+   * @param {number} moveBy Indicates the number of projects to move by. Only allows -1 or 1. 
+   */
+  function moveToProj(moveBy) {
+    console.assert(moveBy === -1 || moveBy === 1, 'Invalid moveBy value');
+  
+    const currProj = projects[projIndex];
+    projIndex = (moveBy + projects.length + projIndex) % projects.length;
+    const nextProj = projects[projIndex];
+  
+    if (moveBy > 0) {
+      showNextProj(currProj, nextProj, Slider.LEFT.hide, Slider.LEFT.show);
+    } else {
+      showNextProj(currProj, nextProj, Slider.RIGHT.hide, Slider.RIGHT.show);
+    }
+  }
+
+  const ANIMATION_DURATION = 1;
```

Comment:
What's this duration, I guess it's in seconds?

Please be more specific in naming, e.g, `ANIMATION_DURATION_MS` or `ANIMATION_DURATION_SECONDS`